allure-pytest/test/conftest.py

```diff
@@ -22,17 +23,23 @@ def inject_matchers(doctest_namespace):
 def _runner(allure_dir, module, *extra_params):
     FNULL = open(os.devnull, 'w')
     extra_params = ' '.join(extra_params)
-    cmd = shlex.split('pytest --alluredir=%s %s %s' % (allure_dir, extra_params, module))
+    cmd = shlex.split('pytest --alluredir=%s %s %s' % (allure_dir, extra_params, module),
+                      posix=False if os.name == "nt" else True)
     subprocess.call(cmd, stdout=FNULL, stderr=FNULL)
 
 
+def path_leaf(path):
+    head, tail = ntpath.split(path)
+    return tail or ntpath.basename(head)
+
+
 @pytest.fixture(scope='module')
 def allure_report_with_params(request, tmpdir_factory):
     module = request.module.__file__
     tmpdir = tmpdir_factory.mktemp('data')
 
     def run_with_params(*params):
-        key = '{thread}{module}{param}'.format(thread=thread_tag(), module=module, param=''.join(params))
+        key = '{thread}{module}{param}'.format(thread=thread_tag(), module=os.path.split(path_leaf(module))[0], param=''.join(params).replace(":",""))
```

Comment:
For now, by keeping only module name (without extension) and removing colons from params the tests run, but I am not sure if this is the best way.